src/cli.rs

```diff
@@ -422,6 +422,12 @@ impl Cli {
         if self.auth_type == AuthType::bearer && self.auth.is_some() {
             self.bearer = self.auth.take();
         }
+        // We would like for --check-status to default to true but that is not
+        // something structopt supports.
+        self.check_status = true;
```

Comment:
What do we think about defaulting to cURL's `--fail` flag in translated commands?

Reply:
It adds noise and it's not even strictly equivalent (curl doesn't fail on 3XX status codes).

I think that ideally we'd add it only if `--check-status` was set explicitly.

We could have a `no_check_status` field that we populate manually, so other code can distinguish between explicitly check/explicitly don't check/default. (I'm not sure that's the cleanest solution.)